graph-rag/app.py

```diff
@@ -1,14 +1,20 @@
-from fastapi import FastAPI, Request
 import os, re
-from fastapi.responses import JSONResponse
-from neo4j import GraphDatabase, AsyncGraphDatabase
-from langchain_community.embeddings import OllamaEmbeddings
-from langchain_community.chat_models import ChatOllama
-
-from prompts import PROMPT_QWEN
+from fastapi import FastAPI, Request
+from neo4j import GraphDatabase
+from langchain_community.embeddings.ollama import OllamaEmbeddings
+from langchain_community.chat_models.ollama import ChatOllama
+from langchain_community.embeddings.openai import OpenAIEmbeddings
+from langchain_community.chat_models.openai import ChatOpenAI
+from prompts import PROMPT_QWEN, PROMPT_OPENAI
 
 app = FastAPI()
 
+### For OpenAI users, please uncomment these following lines:
+# use_openai = os.environ.get('OPENAI_API_KEY', None) is not None
+# if use_openai:
```

Comment:
openAI users have to uncomment this, but the value of `use_openai` is still not set -- should we remove the "if"

Reply:
`use_openai = os.environ.get('OPENAI_API_KEY', None) is not None` I think this is how I set the value for `use_openai`